crypto/comp_methods.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include <openssl/comp.h>
+#include <openssl/obj_mac.h>
+
+#include "internal/cryptlib.h"
+
+#define OSSL_COMP_NULL_IDX       0
+#define OSSL_COMP_ZLIB_IDX       1
+#define OSSL_COMP_NUM_IDX        2
+
+static int sk_comp_cmp(const OSSL_COMP *const *a, const OSSL_COMP *const *b)
```

Comment:
This whole function needs to be inside `#ifdef OPENSSL_NO_COMP` - it is causing the one remaining CI failure.